src/App.vue

```diff
@@ -371,50 +428,139 @@ export default {
             this.itemModal.icon = '';
             this.itemModal.image = '';
             this.itemModal.url = '';
+
+            this.$modal.hide('itemModal');
         },
 
-        openSectionModel(sectionKey) {
-            this.sectionModal.isOpened = true;
+        openSectionModal(sectionKey) {
             this.sectionModal.sectionKey = sectionKey;
 
             if (sectionKey !== undefined) {
                 let section = this.sections[sectionKey];
 
                 this.sectionModal.title = section.title;
             }
+
+            this.$modal.show('sectionModal');
         },
 
         submitSectionModal() {
-            this.sections.push({
-                title: this.sectionModal.title,
-                items: [],
-            });
+            let key = this.sectionModal.sectionKey;
+
+            if (key !== undefined) {
+                this.sections[key].title = this.sectionModal.title;
+            } else {
+                this.sections.push({
+                    title: this.sectionModal.title,
+                    items: [],
+                });
+            }
 
             this.closeSectionModal();
         },
 
+        removeSection(sectionKey) {
+            if(confirm("Delete section?")) {
+                this.sections.splice(sectionKey, 1);
+            }
+        },
+
         closeSectionModal() {
-            this.sectionModal.isOpened = false;
             this.sectionModal.sectionKey = null;
             this.sectionModal.title = '';
+
+            this.$modal.hide('sectionModal');
+        },
+
+        openImport () {
+            this.importExportModal.title = 'Import';
+            this.isImport = true;
+
+            this.$modal.show('importExport');
+        },
+
+        openExport () {
+            let storage = {
+                sections: this.sections,
+                theme: this.theme,
+            };
+
+            this.importExportModal.title = 'Export';
+            this.importExportModal.data = JSON.stringify(storage);
+
+            this.$modal.show('importExport');
+        },
+
+        importJson (newStorage) {
+            let importText = JSON.parse(newStorage);
+
+            if(this.themes.indexOf(importText.theme) !== -1) {
+                this.theme = importText.theme;
+            }
+
+            if(! importText.sections) {
+                return;
+            }
+
+            let sections = importText.sections.filter(section => {
+                return !! section.title;
+            }).map(section => {
+                section.items = section.items.filter(item => {
+                    return item.title && item.url;
+                }).map(item => {
+                    return {
+                        title: item.title,
+                        icon: item.icon || '',
+                        image: item.image || '',
+                        url: item.url,
+                    };
+                });
+
+                return section;
+            });
```

Comment:
Break the logic into new lines:

```js
let sections = importText
    .sections
    .filter(section => !! section.title)
    .map(section => {
        section.items = section
            .items
            .filter(item => item.title && item.url)
            .map(item => ({
                title: item.title,
                icon: item.icon || '',
                image: item.image || '',
                url: item.url,
            }));

        return section;
    });
```

Also take a look on some direct returns I changed, eg on the filter method